open-sphere-plugins/arcgis/src/main/java/io/opensphere/arcgis2/envoy/ArcGISLayerListEnvoy.java

```diff
@@ -102,23 +122,70 @@ public void query(DataModelCategory category, Collection<? extends Satisfaction>
             throw new QueryException("Server URL must be provided in category.");
         }
 
-        URL url;
-        try
+        DataModelCategory localCategory = new DataModelCategory(category.getSource(), category.getFamily(),
```

Comment:
Could 125-138 be factored out into a method?  Maybe it's hard since there's 2 things being set.

Reply:
I tried and failed due to the 2 return values.